### PR TITLE
Allow running on Cockroach CI

### PR DESCRIPTION
**Pull request by justinj**
Still more do to here to get this working (docker?) But here's a preliminary setup.

### CODE REVIEW

File: hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseCoreFunctionalTestCase.java [hunk 1]
```diff
@@ -99,6 +101,7 @@ protected void buildSessionFactory() {
 		buildSessionFactory( null );
 	}
 
+
```
Comment (benesch) [inline hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseCoreFunctionalTestCase.java]: All the changes in this file look spuroius.

File: hibernate-core/src/main/java/org/hibernate/dialect/CockroachDBDialect.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.sql.Types;
+
+import org.hibernate.dialect.function.StandardSQLFunction;
+import org.hibernate.type.StandardBasicTypes;
+
+/**
+ * An SQL dialect for CockroachDB.
+ * Makes no changes at the moment, since CockroachDB now has the type aliases
+ * used. Keeping this file around for the future, though.
```
Comment (benesch) [inline hibernate-core/src/main/java/org/hibernate/dialect/CockroachDBDialect.java]: "Now has the type aliases used" doesn't quite make sense to me. "Now provides type aliases for all types that Hibernate uses"?

File: hibernate-core/src/test/resources/hibernate.properties [hunk 3]
```diff
@@ -24,4 +30,4 @@ hibernate.cache.region.factory_class org.hibernate.testing.cache.CachingRegionFa
 javax.persistence.validation.mode=NONE
 hibernate.service.allow_crawling=false
 hibernate.session.events.log=true
-hibernate.hql.bulk_id_strategy.global_temporary.drop_tables=true
\ No newline at end of file
+hibernate.hql.bulk_id_strategy.global_temporary.drop_tables=true
```
Comment (benesch) [inline hibernate-core/src/test/resources/hibernate.properties]: Did you mean to commit the changes to this file? Seems like you're writing a custom config file in the test harness.

File: cockroach-testing/main_test.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+	scanner := bufio.NewScanner(cmdReader)
+	go func() {
+		for scanner.Scan() {
+			line := scanner.Text()
+			if failed.MatchString(line) {
+				t.Error(line)
+			}
+			fmt.Println(scanner.Text())
+		}
```
Comment (benesch) [inline cockroach-testing/main_test.go]: You probably want to check `scanner.Err()` here.

File: cockroach-testing/main_test.go [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	}()
+
+	err = cmd.Start()
+	if err != nil {
```
Comment (benesch) [inline cockroach-testing/main_test.go]:
```
if err := cmd.Start(); err != nil {
```

(and below, too)

**Review by benesch — APPROVED** [review]
This is awesome! :tada:

~Looks like a script with the `go run` invocation that'll kick off the test is missing. Are you planning to add that in another PR?~ Just saw your PR description.

File: hibernate-core/src/test/resources/hibernate.properties (hunk 3 above)
Comment (justinj) [reply hibernate-core/src/test/resources/hibernate.properties]: This is the global config file, which allows someone to hit "run test" in IntelliJ or whatever and have the tests run against a running Cockroach instance, which can be useful. I'll clean it up and add a comment!

File: hibernate-core/src/main/java/org/hibernate/dialect/CockroachDBDialect.java (hunk 2 above)
Comment (justinj) [reply hibernate-core/src/main/java/org/hibernate/dialect/CockroachDBDialect.java]: 👍

File: hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseCoreFunctionalTestCase.java (hunk 1 above)
Comment (justinj) [reply hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseCoreFunctionalTestCase.java]: Yes indeed!

File: cockroach-testing/main_test.go (hunk 5 above)
Comment (justinj) [reply cockroach-testing/main_test.go]: Done.

File: cockroach-testing/main_test.go (hunk 4 above)
Comment (justinj) [reply cockroach-testing/main_test.go]: Done.

**Review by justinj — COMMENTED** [review]
Thanks for taking a look, I've updated some stuff (and re-added some dialect stuff that I had forgotten was relevant). PTAL? Tamir was telling me that the existing nightly tests don't run the same way as this should - can we talk through that script you mention at some point?

File: hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseUnitTestCase.java [hunk 6]
```diff
@@ -45,6 +49,22 @@
 	@Rule
 	public TestRule globalTimeout = Timeout.millis( TimeUnit.MINUTES.toMillis( 30 ) ); // no test should run longer than 30 minutes
 
+	@Rule
+	public TestName name = new TestName();
+
+  // This was included so that the name of the test running would be included
+  // in the Cockroach logs, will need to figure out a way to get it to work
+  // with the dynamically allocated port for the test server.
```
Comment (benesch) [inline hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseUnitTestCase.java:57]: Indent here looks funny. Tabs vs spaces maybe?

Comment (benesch) [reply hibernate-testing/src/main/java/org/hibernate/testing/junit4/BaseUnitTestCase.java:57]: Also, can you just use "jdbc:default:connection" as your connection URL?

**Pull request closed without merging**